app/src/main-process/log.ts

```diff
@@ -78,12 +78,7 @@ function getLogger(): Promise<winston.LogMethod> {
 
     const logPath = getLogPath()
 
-    Fs.mkdir(logPath, (error) => {
-      if (error && error.code !== 'EEXIST') {
-        reject(error)
-        return
-      }
-
+    fileSystem.mkdirIfNeeded(logPath, (error) => { reject(error) }).then(() => {
```

Comment:
There's a subtle change here - both the `reject` function and the `then` callback will fire. I'll explain and propose another approach further up.